### DISCUSSION TITLE
Manually adding in system /station

### DISCUSSION THREAD

**Discussion by Falcon911** [Fi0x/EDCT#131] [Ideas]
Sometime manually adding in a station name or system name a searching.

**Comment by Fi0x** [comment]
That's an interesting idea, but I'm not yet sure how this would benefit the search.

Currently all station-markets are loaded from inara and while the tool runs, every new information from the data-network is received as well.
Can you specify a situation where you would need this feature?

**Comment by Falcon911** [comment]
I was doing this current CG and I know it's not a huge money maker per sea. So I was a T'u Tru (whatever it was called) and did not see it with Copper or Gallium materials in your tool.  So I was I was busy writing the reddit and discord notifications that I'm loading/unloading  at X  system and station. When I found your Reddit and Discord buttons in the tool. I was like..wow this would be nice to have if we could manually add in the current inara prices and such and push that out.

**Reply by Fi0x** [reply]
So it would be sufficient for you if there would be an extra UI-part that won't interact with the actual database?

This part only needs to have Text fields for `System Name`, `Station Name`, `Commodity Name` and buttons for the "Buy" and "Sell" Texts.
With the given information it would then create the same Texts that you get when you press the reddit button.

Posslible Layout might look like this: 
![image](https://user-images.githubusercontent.com/58072122/206166569-7e35a649-c866-4fbc-ad96-ca56b6b1eadc.png)

Button to reach new Window could be on the bottom of the settings:
![image](https://user-images.githubusercontent.com/58072122/206167927-a9f6ec97-483f-4dd1-b6e8-ac27103cc0fc.png)

Did I understand you correctly, or is there something else you would like to see or change in the picture above?

**Reply by Falcon911** [reply]
Yes you have the idea ... Thats perfect. Maybe add Ship name in there too. Can we have a drop downlist of the Commodity's ? hmm.. Would it be too much to have a data call that once we choose a station that the commodity can self populate due to the fact that some  stations do not have certain Commodity's ? Maybe that's too much network data to a from the inara or Eddb...  I can get a csv file with all the current Commidity's made up for ya if you need it.

**Reply by Fi0x** [reply]
> Maybe add Ship name in there too.

I would just use the Carrier name from the settings, or did you mean another ship name?

> Can we have a drop downlist of the Commodity's ?

I already thought about drop-down and I'll probably use it for the commodity-selection.

> Would it be too much to have a data call that once we choose a station that the commodity can self populate due to the fact that some stations do not have certain Commodity's ?

Receiving a list of valid commodities for a single station is only one call, so that shouldn't be a problem. However this can only be done after the station-name and system-name are provided. So I think a search-field at the top of the drop-down might be the easiest solution. Something like the commodity selection on inara: 
![image](https://user-images.githubusercontent.com/58072122/206231604-ae53ec35-f08d-4a19-b691-8844ed44dcd2.png)

> I can get a csv file with all the current Commidity's made up for ya if you need it.

Don't worry, that information is already in each local database and easily available within the tool.

**Reply by Falcon911** [reply]
Looks good. Would make things easier.
